### PR TITLE
GH-241: Update docs

### LINKED ISSUE #241: Ensure docs are up-to-date
Need to take a scan through the [current docs](https://docs.spring.io/stream-applications/docs/current/reference/html/index.html#applications/) and make sure that the content is up-to-date w/ the latest things in `2021.1.2`. 

- [x] reference guide
- [x] https://spring.io/projects/spring-cloud-stream-applications#overview
  - [x] update links from `master` -> `main`
  - [x] fix "app registration" link formatting
  - [x] fix "Step 5" of `quickstart` (needs `--server.port=0`)

### PR DESCRIPTION
**Pull request by onobc**
This updates the reference guide (links, versions, etc..)

Fixes #241

1. I scoured through the ref guide (line by lline) and made sure all links work and are relevant etc..
2. I built locally
3. I went through the generated ref guide (line by line) and verified the changes in step 1)

| ℹ️ I deleted the pom-dependencies page (and its associated images) as it was not used and referenced the old app starters and I don't see us ever wanting to show that w/o large updates |
| --- |

### CODE REVIEW

File: stream-applications-release-train/stream-applications-docs/src/main/asciidoc/index.adoc [hunk 1]
```diff
@@ -15,13 +15,15 @@ Sabby Anandan; Artem Bilan; Marius Bogoevici; Eric Bottard; Mark Fisher; Ilayape
...
+:spring-core-version: 5.3.20
+
+:scst-core-version: 3.2.0
 
 ifdef::backend-html5[]
 
 Version {project-version}
 
-(C) 2012-2021 Pivotal Software, Inc.
+(C) 2012-2022 Pivotal Software, Inc.
```
Comment (artembilan) [inline stream-applications-release-train/stream-applications-docs/src/main/asciidoc/index.adoc:26]: I guess this one has to be now like `VMware, Inc. All rights reserved.`

File: stream-applications-release-train/stream-applications-docs/src/main/asciidoc/index-docinfo.xml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <productname>Spring Cloud Data Flow</productname>
 <releaseinfo>{project-version}</releaseinfo>
 <copyright>
-	<year>2013-2021</year>
+	<year>2013-2022</year>
 	<holder>Pivotal Software, Inc.</holder>
```
Comment (artembilan) [inline stream-applications-release-train/stream-applications-docs/src/main/asciidoc/index-docinfo.xml:5]: As well as this one.

Not sure though, that this file is used anywhere...

@sobychacko ?

**Review by artembilan — CHANGES_REQUESTED** [review]

File: stream-applications-release-train/stream-applications-docs/src/main/asciidoc/index.adoc (hunk 1 above)
Comment (onobc) [reply stream-applications-release-train/stream-applications-docs/src/main/asciidoc/index.adoc:26]: Yep, it crossed my mind as well. It is displayed on [index page at top](https://docs.spring.io/stream-applications/docs/current/reference/html/). 

Other ref docs looks like they have the copyright date but have no company name (ex: [Spring-Boot](https://docs.spring.io/spring-boot/docs/current/reference/html/legal.html#legal)).

I will remove the name portion out of this.

Comment (onobc) [reply stream-applications-release-train/stream-applications-docs/src/main/asciidoc/index.adoc]: Done

File: stream-applications-release-train/stream-applications-docs/src/main/asciidoc/index-docinfo.xml (hunk 2 above)
Comment (onobc) [reply stream-applications-release-train/stream-applications-docs/src/main/asciidoc/index-docinfo.xml]: I updated this to be `VMware, Inc` as I am not sure if this is used anywhere or not - so just made it accurate.

**Review by artembilan — APPROVED** [review]
Will leave the final merge decision for @sobychacko .

Thanks

File: stream-applications-release-train/stream-applications-docs/src/main/asciidoc/index-docinfo.xml (hunk 2 above)
Comment (sobychacko) [reply stream-applications-release-train/stream-applications-docs/src/main/asciidoc/index-docinfo.xml]: I think this file is a remnant from some previous times. We can probably safely delete this for the 2022 release.

**Comment by sobychacko** [comment]
All look good from my side.

**Review by sobychacko — APPROVED** [review]

**Pull request merged**